autoload/jedi.vim

```diff
@@ -392,20 +461,61 @@ endfunction
 " helper functions
 " ------------------------------------------------------------------------
 
-function! jedi#add_goto_window(len) abort
-    set lazyredraw
-    cclose
+" a:mode: 'goto' or 'usages'
+function! jedi#add_goto_window(mode, len, ...) abort
+    let select_entry = a:0 ? a:1 : 0
     let height = min([a:len, g:jedi#quickfix_window_height])
-    execute 'belowright copen '.height
-    set nolazyredraw
-    if g:jedi#use_tabs_not_buffers == 1
-        noremap <buffer> <CR> :call jedi#goto_window_on_enter()<CR>
-    endif
-    augroup jedi_goto_window
-      au!
-      au WinLeave <buffer> q  " automatically leave, if an option is chosen
-    augroup END
-    redraw!
+
+    " Using :cwindow allows to stay in the current window in case it is opened
+    " already.
+    execute 'belowright cwindow '.height
+
+    if &filetype ==# 'qf'
+        " Setup quickfix window, but only if it was opened / initially.
+        if g:jedi#use_tabs_not_buffers == 1
+            noremap <buffer> <CR> :call jedi#goto_window_on_enter()<CR>
+        endif
+
+        augroup jedi_goto_window
+            if a:mode ==# 'goto'
+                autocmd WinLeave <buffer> q  " automatically leave, if an option is chosen
+            else
+                autocmd BufWinLeave <buffer> call jedi#clear_usages()
+            endif
+        augroup END
+
+        " Preview
+        " exe "nnoremap <buffer> p \<CR>:PythonJedi jedi_vim.highlight_usages_for_vim_win()\<CR>\<C-w>p"
```

Comment:
I guess this was meant for Vim to being able to highlight (from the quickfix window), since unlike Neovim highlights `matchaddpos()` only works for the current window.
Useful?

By now Vim has text-properties, which could/should be used probably instead of `matchaddpos()`.

Reply:
Was this in there before? IMO it's not necessary for now.